neo/Core/StateDescriptor.cs

```diff
@@ -0,0 +1,155 @@
+            Value = reader.ReadVarBytes(65535);
+            switch (Type)
+            {
+                case StateType.Account:
+                    CheckAccountState();
+                    break;
+                case StateType.Validator:
+                    CheckValidatorState();
+                    break;
+            }
+        }
+
+        private Fixed8 GetSystemFee_Validator()
+        {
+            switch (Field)
+            {
+                case "Registered":
+                    if (Value.Any(p => p != 0))
+                        return Fixed8.FromDecimal(1000);
+                    else
+                        return Fixed8.Zero;
+                default:
+                    throw new InvalidOperationException();
+            }
+        }
+
+        void ISerializable.Serialize(BinaryWriter writer)
+        {
+            writer.Write((byte)Type);
+            writer.WriteVarBytes(Key);
+            writer.WriteVarString(Field);
+            writer.WriteVarBytes(Value);
+        }
+
+        public JObject ToJson()
+        {
+            JObject json = new JObject();
+            json["type"] = Type;
+            json["key"] = Key.ToHexString();
+            json["field"] = Field;
+            json["value"] = Value.ToHexString();
+            return json;
+        }
+
+        internal bool Verify()
+        {
+            switch (Type)
+            {
+                case StateType.Account:
+                    return VerifyAccountState();
+                case StateType.Validator:
+                    return VerifyValidatorState();
+                default:
+                    return false;
+            }
+        }
+
+        private bool VerifyAccountState()
+        {
+            switch (Field)
+            {
+                case "Votes":
+                    if (Blockchain.Default == null) return false;
+                    ECPoint[] pubkeys;
+                    try
+                    {
+                        pubkeys = Value.AsSerializableArray<ECPoint>((int)Blockchain.MaxValidators);
+                    }
+                    catch (FormatException)
+                    {
+                        return false;
+                    }
+                    UInt160 hash = new UInt160(Key);
+                    AccountState account = Blockchain.Default.GetAccountState(hash);
+                    if (account?.IsFrozen != false) return false;
+                    if (pubkeys.Length > 0)
+                    {
+                        if (account.GetBalance(Blockchain.GoverningToken.Hash).Equals(Fixed8.Zero)) return false;
+                        HashSet<ECPoint> sv = new HashSet<ECPoint>(Blockchain.StandbyValidators);
+                        DataCache<ECPoint, ValidatorState> validators = Blockchain.Default.GetStates<ECPoint, ValidatorState>();
+                        foreach (ECPoint pubkey in pubkeys)
+                            if (!sv.Contains(pubkey) && validators.TryGet(pubkey)?.Registered != true)
```

Comment:
maybe this should be `or` ?
` if (!sv.Contains(pubkey) || validators.TryGet(pubkey)?.Registered != true)`

Reply:
It is `and`.